### ISSUE TITLE
Propose: make block-spacing not report about comment just after open curly brace

### ISSUE THREAD

**Issue by Mingun** [eslint/eslint#9439]
**Tell us about your environment**

* **ESLint Version:** 3.12.0
* **Node Version:** 5.12.0
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**
`gulp-eslint` if I correclty understand this question

**Please show your full configuration:**
It isn't sure that you mean, a configuration for project https://github.com/pegjs/pegjs, probably https://github.com/pegjs/pegjs/blob/master/.eslintrc.json

**What did you do? Please include the actual source code causing the issue.**
Example
```js
function someFunction() {// Some brief comment about function
  ...
}
```

**What did you expect to happen?**
I expect, that rule `block-spacing` not trigger warning because I think this is very common pattern for comments. Adding space between `{` and `//` seems ugly.

**What actually happened? Please include the actual, raw output from ESLint.**
I specially make change to trigger error in the following code:
```js
      while (ip < end) {// line 607 in lib\compiler\passes\generate-js.js
```
```cmd
D:\Projects\parsers\pegjs>gulp
[17:22:38] Using gulpfile D:\Projects\parsers\pegjs\gulpfile.js
[17:22:38] Starting 'default'...
[17:22:38] Starting 'lint'...
[17:22:50]
D:\Projects\parsers\pegjs\lib\compiler\passes\generate-js.js
  607:24  error  Requires a space after '{'  block-spacing

✖ 1 problem (1 error, 0 warnings)

[17:22:51] 'lint' errored after 13 s
[17:22:51] ESLintError in plugin 'gulp-eslint'
Message:
    Failed with 1 error
[17:22:51] 'default' errored after 13 s
[17:22:51] ESLintError in plugin 'run-sequence(lint)'
Error
    at finish (D:\Projects\parsers\pegjs\node_modules\run-sequence\index.js:56:13)
    at Gulp.onError (D:\Projects\parsers\pegjs\node_modules\run-sequence\index.js:67:4)
    at emitOne (events.js:95:20)
    at Gulp.emit (events.js:182:7)
    at Gulp.Orchestrator._emitTaskDone (D:\Projects\parsers\pegjs\node_modules\gulp\node_modules\orchestrator\index.js:264:8)
    at D:\Projects\parsers\pegjs\node_modules\gulp\node_modules\orchestrator\index.js:275:23
    at finish (D:\Projects\parsers\pegjs\node_modules\gulp\node_modules\orchestrator\lib\runTask.js:21:8)
    at Transform.<anonymous> (D:\Projects\parsers\pegjs\node_modules\gulp\node_modules\orchestrator\lib\runTask.js:52:4)
    at Transform.f (D:\Projects\parsers\pegjs\node_modules\gulp\node_modules\orchestrator\node_modules\end-of-stream\node_modules\once\once.js:17:25)
    at emitOne (events.js:95:20)

D:\Projects\parsers\pegjs>
```

**Comment by VictorHom** [comment]
@Mingun I understand the issue you opened and can recreate. Currently, the rule is working as expected. I think if this optional feature is supported, it would apply to Block comments as well. That means the following would be valid too.
```
function someFunction() {/* Some brief comment about function*/
  ...
}
```
I'm not too familiar with how common this way of commenting is (with space vs without space before the comment.), so I will let @eslint/eslint-issues-team weigh in

**Comment by not-an-aardvark** [comment]
In my opinion, this seems like a very personal preference, so it might be a better fit for a [custom rule](https://eslint.org/docs/developer-guide/working-with-rules) rather than a modification to the core `block-spacing` rule. But I'm not strongly opposed to adding an option for it if there are other team members in favor.

**Comment by ilyavolodin** [comment]
I agree with @not-an-aardvark This seems like a very personal preference, and better suited to a plugin, then core option. Closing this issue, since a few members of the team feel this way. Feel free to open a new one, if you do not agree, and can provide a few styleguides that require this pattern.
